pkg/eestream/decode.go

```diff
@@ -47,25 +52,29 @@ func (dr *decodedReader) Read(p []byte) (n int, err error) {
 		// the channel has a buffer size to contain all the errors
 		// even if we read none, so we can return without receiving
 		// every channel value
-		errs := make(chan error, len(dr.rs))
+		errs := make(chan readerError, len(dr.rs))
 		for i := range dr.rs {
 			go func(i int) {
 				// fill the buffer from the piece input
 				_, err := io.ReadFull(dr.rs[i], dr.inbufs[i])
-				errs <- err
+				errs <- readerError{i, err}
 			}(i)
 		}
 		// catch all the errors
+		inbufs := make(map[int][]byte)
```

Comment:
you can reduce memory allocations by having this be 
```
inbufs := make(map[int][]byte, len(dr.inbufs))
```

this tells the new map it will have to store `len(dr.inbufs)` things at most, so it can allocate that memory upfront instead of growing every time a new map element is added

Reply:
yes, makes sense.